src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+        FullscreenOperator {target,scene,size_original,position_original,parent_original}.init()
+    }
+
+    fn init(self) -> Self {
+        let original_pos = self.target.display_object().global_position();
+
+        // Change parent
+        self.target.display_object().set_parent(self.scene.display_object());
+        self.target.unset_layers_all(&self.scene);
+        set_layers_fullscreen(&self.target, &self.scene);
+
+        let margin = 0.0;
+        let scene_shape = self.scene.shape();
+        let size_new    = Vector3::new(scene_shape.width(), scene_shape.height(),0.0) * (1.0 - margin);
+
+        // TODO Currently we assume objects are center aligned, but this needs to be properly
+        // accounted for here.
+
+        let frp_network      = &self.target.network().clone_ref();
+        let target_pos       = Vector3::zero();
+        let original_size    = self.size_original;
+        let target_size      = size_new;
+        let target           = self.target.clone_ref();
+        let resize_animation = animation(frp_network, move |value| {
+            let pos  = original_pos  * (1.0 - value) + target_pos * value;
+            let size = original_size * (1.0 - value) + target_size * value;
+            target.set_position(pos);
+            target.set_size(size);
+        });
+        resize_animation.set_target_position(1.0);
+
+        self.scene.views.toggle_overlay_cursor();
+
+        self
+    }
+
+    /// Undo the fullscreen operation and restore the previous state exactly as it was.
+    pub fn undo(self) {
+        let global_pos_start = self.target.global_position();
+
+        self.target.unset_layers_all(&self.scene);
+        set_layers_normal(&self.target, &self.scene);
+
+        if let Some(parent) = self.parent_original.as_ref() {
+            self.target.display_object().set_parent(&parent);
+        }
+
+        let parent_pos     = self.parent_original.map(|p| p.global_position());
+        let parent_pos     = parent_pos.unwrap_or_else(Vector3::zero);
+        let mut source_pos = self.target.position();
+        source_pos        += global_pos_start ;
+        source_pos        -= parent_pos ;
+        let source_pos     = source_pos;
+
+        self.target.set_position(source_pos);
+
+        let original_pos     = self.target.position();
+        let target_pos       = self.position_original;
+        let original_size    = self.target.size();
+        let target_size      = self.size_original;
+        let target           = self.target.clone_ref();
+        let frp_network      = &self.target.network().clone_ref();
+        let resize_animation = animation(frp_network, move |value| {
+            let pos  = original_pos  * (1.0 - value) + target_pos * value;
+            let size = original_size * (1.0 - value) + target_size * value;
+            target.set_position(pos);
+            target.set_size(size);
+        });
+        resize_animation.set_target_position(1.0);
```

Comment:
This is very unsafe. What happens here is complex:

1. You define FRP network above and you move to lambda `target`, which has interior mutability.
2. You are starting the FRP network here, which means the lambda could be evaluated. In particular, `target.set_position` or something could be fired. Other things could listen to this frp network and could potentially trigger undoing or doing fullscreen.
3. If they do so, there ow uldbe runtime crash because you are now in non-interial mutable thing (you have mutable borrowed it).

Thats why we often use `RefCells` only on some fields, not on whole structs like here - just to borrow things for the shortest possible time. If you borrow thing and you eval method on borrowed thing, you are not allowed to run any FRP that changes things that can be observed from outside of this borrow.